test-vectors.md

```diff
@@ -152,43 +152,46 @@ Parameters:
 Format:
 
 ```
-struct {
-  MLSPlaintext commit;       // chosen by generator.  membership_tag and
-                             // confirmation_tag MUST be valid; otherwise 
-                             // content is only used for transcript.
-  CryptoValue tree_hash;     // chosen by generator
-
-  CryptoValue commit_secret; // chosen by generator
-  CryptoValue psk_secret;    // chosen by generator
-
-  CryptoValue confirmed_transcript_hash;
-  CryptoValue interim_transcript_hash;
-  CryptoValue group_context;
-
-  CryptoValue joiner_secret;
-  CryptoValue welcome_secret;
-  CryptoValue epoch_secret;
-  CryptoValue init_secret;
-
-  CryptoValue sender_data_secret;
-  CryptoValue encryption_secret;
-  CryptoValue exporter_secret;
-  CryptoValue authentication_secret;
-  CryptoValue external_secret;
-  CryptoValue confirmation_key;
-  CryptoValue membership_key;
-  CryptoValue resumption_secret;
-
-  HPKEPublicKey external_pub;
-} Epoch;
-
-struct {
-  CipherSuite cipher_suite;
-  CryptoValue group_id;
-  CryptoValue initial_tree_hash;    // chosen by generator
-  CryptoValue initial_init_secret;  // chosen by generator
-  Epoch epochs<0..2^32-1>;
-} KeyScheduleTestVector;
+{
+  "cipher_suite": /* uint16 */,
+  "group_id": /* hex-encoded binary data */,
+  "initial_tree_hash": /* hex-encoded binary data */,
+  "initial_init_secret": /* hex-encoded binary data */,
+  "epochs": [
+    {
+      // These fields are chosen by the generator.  The "commit" field contains
+      // TLS-serialized MLSPlaintext object.  The membership_tag and
+      // confirmation_tag in the MLSPlaintext MUST be valid; otherwise its
+      // content is only used for the transcript.
+      "commit": /* hex-encoded binary data */,
+      "tree_hash": /* hex-encoded binary data */,
+      "commit_secret": /* hex-encoded binary data */,
+      "psk_secret": /* hex-encoded binary data */,
+      
+      "confirmed_transcript_hash": /* hex-encoded binary data */,
+      "interim_transcript_hash": /* hex-encoded binary data */,
+      "group_context": /* hex-encoded binary data */,
+      
+      "joiner_secret": /* hex-encoded binary data */,
+      "welcome_secret": /* hex-encoded binary data */,
+      "epoch_secret": /* hex-encoded binary data */,
```

Comment:
The epoch secret is a temporary value in the key schedule that should never get exposed. We can keep it in here but it requires extra code for testing and should always be tested through all the secrets that follow here. I'd remove it here.